packages/compass-data-modeling/src/components/export-diagram-modal.tsx

```diff
@@ -59,20 +63,71 @@ const ExportDiagramModal = ({
   model,
   onCloseClick,
 }: ExportDiagramModalProps) => {
-  const [exportFormat, setExportFormat] = useState<'json' | null>(null);
-
-  const onExport = useCallback(() => {
-    if (!exportFormat || !model) {
-      return;
+  const [exportFormat, setExportFormat] = useState<'png' | 'json' | null>(null);
+  const diagram = useDiagram();
+  const [isExporting, setIsExporting] = useState(false);
+  const abortControllerRef = useRef<AbortController | null>(null);
+  const toast = useToast();
```

Comment:
Use our openToast instead, not sure how are we even exporting this one, we really shouldn't